src/lib.rs

```diff
@@ -106,11 +130,87 @@ impl SimpleMTKCore for SimpleMTK {
         self.token_id = self.token_id.saturating_add(1);
     }
 
+    /// Burns a token.
+    ///
+    /// Requirements:
+    /// * sender MUST have sufficient amount of token.
+    ///
+    /// Arguments:
+    /// * `id`: Token ID.
+    /// * `amount`: Token's amount to be burnt.
     fn burn(&mut self, id: TokenId, amount: u128) {
         MTKCore::burn(self, vec![id], vec![amount]);
         let sup = self.supply(id);
         let mut _balance = self
             .supply
             .insert(self.token_id, sup.saturating_sub(amount));
     }
+
+    /// Transforms FT tokens to multiple NFTs.
+    ///
+    /// Requirements:
+    /// * a sender MUST have sufficient amount of tokens to burn,
+    /// * a sender MUST be the owner.
+    ///
+    /// Arguments:
+    /// * `id`: Token ID.
+    /// * `amount`: Token's amount to be burnt.
+    /// * `accounts`: To which accounts to mint NFT.
+    /// * `nft_ids`: NFTs' IDs to be minted.
+    /// * `nfts_metadata`: NFT's metadata.
+    fn transform(
+        &mut self,
+        id: TokenId,
+        amount: u128,
+        accounts: Vec<ActorId>,
+        mut nft_ids: Vec<Vec<TokenId>>,
+        nfts_metadata: Vec<Vec<Option<TokenMetadata>>>,
+    ) {
+        // prechecks
+        if nft_ids.len() != nfts_metadata.len() {
+            panic!("MTK: number of ids, accounts and metadata pieces MUST be the same.");
+        }
+        let mut nft_count = 0;
+        for (_, ids) in nft_ids.iter().enumerate() {
+            nft_count += ids.len();
+        }
```

Comment:
Why to enumerate and to not use an index?
```suggestion
        for ids in &nft_ids {
            nft_count += ids.len();
        }
```